docs/source/development.rst

```diff
@@ -0,0 +1,371 @@
+
+Common development commands:
+
+.. code-block:: console
+
+   # Using uv run (recommended):
+   $ uv run FABulous -h           # run CLI with project dependencies
+   $ uv run pytest               # run test suite
+   $ uv run pytest -k test_name  # run specific test
+   $ uv run ruff check           # lint code
+   $ uv run ruff format          # format code
+
+   # Or with activated environment:
+   (.venv) $ FABulous -h
+   (.venv) $ pytest
+   (.venv) $ ruff check
+   (.venv) $ ruff format
+
+Dependency management:
+
+.. code-block:: console
+
+   $ uv add <package>            # add runtime dependency
+   $ uv add --group dev <package> # add development dependency
+   $ uv remove <package>         # remove dependency
+   $ uv lock                     # refresh lock file after manual edits
+
+.. _pre_commit:
+
+Pre-commit Hooks
+----------------
+
+We use `pre-commit <https://pre-commit.com/>`_ hooks to maintain code quality.
+These hooks automatically run formatters and linters before each commit.
+
+Install pre-commit hooks:
+
+.. code-block:: console
+
+   $ uv run pre-commit install
+
+The hooks will now run automatically on ``git commit``. You can also run them manually:
+
+.. code-block:: console
+
+   $ uv run pre-commit run --all-files
+
+If you need to bypass the hooks temporarily (not recommended):
+
+.. code-block:: console
+
+   $ git commit --no-verify
+
+.. _task_automation:
+
+Task Automation with Taskipy
+-----------------------------
+
+FABulous includes pre-configured `taskipy <https://github.com/taskipy/taskipy>`_ tasks to streamline common development and workflow tasks. After setting up the development environment, you can run these tasks using ``task <task-name>``.
+
+Development and Quality Tasks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: console
+
+   $ task format          # Format code with ruff
+   $ task lint            # Lint and fix code issues + run pre-commit
+   $ task check           # Check code without fixing
+   $ task qa              # Run format and check in sequence
+   $ task pre-commit      # Run format and check (for pre-commit hooks)
+   $ task ci-check        # Full CI check (format, lint, test, docs)
+   $ task install-dev     # Install development dependencies
+   $ task clean-all       # Clean all build artifacts and cache files
+
+Documentation Tasks
+~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: console
+
+   $ task docs-setup      # Setup documentation environment
+   $ task docs-apidoc     # Generate API documentation only
+   $ task docs-build      # Generate API docs + build documentation
+   $ task docs-serve      # Serve docs with live reload for development
+   $ task docs-clean      # Clean documentation build artifacts
+
+Project Creation and Setup
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: console
+
+   $ task fab-proj               # Create demo project
+
+FABulous Workflow Tasks
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: console
+
+   # Fabric generation and simulation
```

Comment:
The tasks are nice, but I think in the future we should switch to something like [taskfiles](https://taskfile.dev/) to also replace the makefiles and unify this a bit. 
Since have now something like 3 or 4 ways to run a simulation or generate the fabric, which can lead to confusion for users.